### ISSUE TITLE
Using streamlit for web app GUI - extending the custom server, tensorboard, opencv displays

### ISSUE THREAD

**Issue by Twenkid** [dobkeratops/convnet_stuff#3]
I've just tried the library: 

https://streamlit.io/
https://docs.streamlit.io/library/get-started/create-an-app
https://docs.streamlit.io/library/api-reference

Besides Tensorboard, I've run it a few times with GANs, but I haven't applied it in my projects.
TB also has some input widgest, but this one seems more general-purpose and flexible for web apps: https://github.com/tensorflow/tensorboard

![image](https://user-images.githubusercontent.com/23367640/193709961-4f7c115e-e568-44fc-bef7-a8382141b678.png)

```python
import streamlit as st
import pandas as pd

def main():
  st.write("""# APP STREAMLIST!
  Web Data *Science*!""")
  df = pd.read_csv("z:\\d2.csv")
  st.line_chart(df)

main()

```

d2.csv
```
1
20
45
221
12
45
67
111
5
```

The lib may need to upgrade Python to 3.10.6+

**Comment by dobkeratops** [comment]
still doing this 'adhoc' for the minute, I'm sure these options have solved some problems and are way better presented. it would definitely be nice to have a web UI, even able to pause, snapshot, change vars like learning rate/shortcut etc.

At least I should be able to separate the visualiser off so its easy to swap out
